src/Consumer.ts

```diff
@@ -361,6 +369,20 @@ export class Consumer extends EventEmitter {
     this._retryTimer = setTimeout(this._connect, delay)
   }
 
+  /**
+   * Starts the consumer if it is currently stopped.
+   * When created with lazy=true, begin consuming.
+   */
+  public async start(): Promise<void> {    
+    if (this._readyState !== READY_STATE.CLOSED) {      
+      return
+    }
+
+    this._readyState = READY_STATE.CONNECTING      
+    this._connect()
+    await this._pendingSetup
```

Comment:
This may not do what you expect. The `_pendingSetup` promise will resolve if there happens to be a temporary connection problem (and then we retry). Which means `start()` can return before the connection is actually ready. For this reason you have a choice:
- Wait for the "ready" event (see the end of `_setup()`) instead of `_pendingSetup`. But what if the connection is never established? You also need to stop waiting if a corresponding `close()` is called so you don't get locked up when shutting down an application which can't connect to rabbit (for whatever reason). This is getting complicated!
- Delete `await this._pendingSetup`. Make this a regular sync function.

Reply:
I have done several tests to check the connection was ready, with the ready and error events, but did not get a consistent state. After the ready event was emitted the State of the Consumer was still on Connecting in some cases. Because of this I have added the await for the pendingSetup so the start() method only returns if the setup was complete.

**"You also need to stop waiting if a corresponding close() is called so you don't get locked up when shutting down an application which can't connect to rabbit (for whatever reason). This is getting complicated!"**
=> As the close() method also waits for the pendingSetup, the promise should be resolve if you call close(). I can add a unit test for this case.

**Delete await this._pendingSetup. Make this a regular sync function.**
=> If this is your preferred way, I can change the PR.

Reply:
I think synchronous is best

Reply:
I have updated the PR and make the method synchronous. 👍